#8334 Review mapType configuration and docs (#9012)
geosolutions-it/MapStore2: docs/developer-guide/mapstore-migration-guide.md

The map configuration stores the information related to the visualization mode 2D or 3D after saving a map.
This update include also following changes:

> - `maptype` configuration inside the initialState of localConfig needs to be removed in favor of the global mapType configuration. The global mapType configuration is only needed for project with custom map library settings.

Reviewer: Please remove the sectuon below and add a URL here
```suggestion
- `maptype` configuration inside the initialState of `localConfig.json` needs to be removed. If you were using this configuration to customize the map library, refer to [this documentation](url) to update your configuration.
```

> - `maptype` default state configuration inside the initialState of localConfig.json needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the localConfig.json file, see documentation [here](local-config.md#application-configuration).

Reviewer: ```suggestion
- `maptype` default state configuration inside the initialState of localConfig.json needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the `localConfig.json` file, see documentation [here](local-config.md#application-configuration).
```

> - `maptype` default state configuration inside the initialState of localConfig.json needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the localConfig.json file, see documentation [here](local-config.md#application-configuration).

Reviewer: ```suggestion
- `maptype` default state configuration inside the initialState of `localConfig.json` needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the localConfig.json file, see documentation [here](local-config.md#application-configuration).
```

> - `maptype` default state configuration inside the initialState of localConfig.json needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the localConfig.json file, see documentation [here](local-config.md#application-configuration).

Reviewer: ```suggestion
- `maptype` default state configuration inside the initialState of `localConfig.json` needs to be removed. If a MapStore project needs a particular setup (eg. use only OpenLayers for 2D maps, initialize the app in 3D, ...) it is possible to override the default map libraries configuration with the new `mapType` property in the `localConfig.json` file, see documentation [here](local-config.md#application-configuration).
```

    // ...
}
```
an example of global mapType configuration where OpenLayers is used for mobile and desktop devices:

!!! note:
    The downstream projects based on the MapStore product don't need this update

```diff
{
    // ...
+    "mapType": {

> +       "defaultVisualizationMode": "2D",
> +       "visualizationModes": {
> +           "2D": {
> +               "desktop": "openlayers",
> +               "mobile": "openlayers"
> +           },
> +           "3D": {
> +               "desktop": "cesium",
> +               "mobile": "cesium"
> +           }
> +       }

Reviewer: ```suggestion
```
